ptype/inference.py

```diff
@@ -0,0 +1,319 @@
+    Args:
+        data: Pandas DataFrame of flattened pressure level data.
+        surface_data: Pandas DataFrame of flattened surface data.
+        pressure_levels: List of pressure levels from model.
+        height_levels: Dictionary of height levels (low, high, interval)
+
+    Returns:
+        Pandas Dataframe of interpolated data at height above the surface.
+    """
+    cols = {}
+    height_levels = np.arange(start=height_levels["low"],
+                              stop=height_levels["high"] + height_levels["interval"],
+                              step=height_levels["interval"])
+    for var in ['t', 'dpt', 'u', 'v', 'hgt_above_sfc']:
+        cols[var] = [f"{var}_{int(x)}" for x in pressure_levels]
+
+    var_arrays = []
+    variables = ['t', 'dpt', 'u', 'v']
+    surface_variables = ['t2m', 'd2m', 'u10', 'v10']
+    height_data = data[cols['hgt_above_sfc']].values
+
+    for v, sv in zip(variables, surface_variables):
+        pressure_level_data = data[cols[v]].values
+        height_interp_data = interpolate(height_data, pressure_level_data, height_levels)
+        height_interp_data[:, 0] = surface_data[sv]
+        var_arrays.append(height_interp_data)
+
+    all_data = np.concatenate(var_arrays, axis=1)
+
+    return all_data
+
+
+@jit(nopython=True, parallel=True, cache=True)
+def interpolate(x, y, height_levels):
+
+    arr = np.zeros(shape=(x.shape[0], len(height_levels)))
+
+    for i in numba.prange(arr.shape[0]):
+        arr[i] = np.interp(x=height_levels,
+                           xp=x[i],
+                           fp=y[i])
+    return arr
+
+def transform_data(input_data, transformer):
+    """
+    Transform data for input into ML model.
+    Args:
+        input_data: Pandas Dataframe of input data
+        transformer: Bridgescaler object used to fit data.
+
+    Returns:
+        Pandas dataframe of transformed input.
+    """
+    transformed_data = transformer.transform(pd.DataFrame(input_data, columns=transformer.x_columns_))
+
+    return transformed_data.values
+
+
+def load_model(model_path, model_file, input_scaler_file, output_scaler_file):
+    """
+    Load ML model and bridgescaler object.
+    Args:
+        model_path: Path to ML model.
+
+    Returns:
+        Loaded Tensorflow model, bridgescaler object
+    """
+    config = os.path.join(model_path, "model.yml")
+    with open(config) as cf:
+        conf = yaml.load(cf, Loader=yaml.FullLoader)
+        conf['batch_size'] = 1000
+
+    x_transformer = load_scaler(os.path.join(model_path, "scalers", input_scaler_file))
+    with open(os.path.join(model_path, "scalers", output_scaler_file)) as f:
+        output_scaler = json.load(f)
+    model = CategoricalDNN(**conf["model"])
+    model.build_neural_network(len(x_transformer.x_columns_), len(output_scaler['classes_']))
+    model.model.load_weights(os.path.join(model_path, "models", model_file))
+
+    return model, x_transformer
+
+
+def grid_preditions(data, preds):
```

Comment:
spelling error. Should be grid_predictions

Reply:
resolved